src/StoreApi/Utilities/ProductQueryFilters.php

```diff
@@ -47,6 +47,72 @@ public function get_filtered_price( $request ) {
 		return $wpdb->get_row( $price_filter_sql ); // phpcs:ignore
 	}
 
+	/**
+	 * Get stock status counts for the current products.
+	 *
+	 * @param \WP_REST_Request $request The request object.
+	 * @return array status=>count pairs.
+	 */
+	public function get_stock_status_counts( $request ) {
+		global $wpdb;
+		$product_query         = new ProductQuery();
+		$stock_status_options  = array( 'instock', 'outofstock', 'onbackorder' );
```

Comment:
```suggestion
		$stock_status_options  = array_keys( wc_get_product_stock_status_options() );
```

If you make this change, it will allow us to get custom stock statuses.